code/modules/mob/living/simple_animal/hostile/mining_mobs.dm

```diff
@@ -374,11 +374,16 @@
 
 /mob/living/simple_animal/hostile/asteroid/goliath/OpenFire()
 	var/tturf = get_turf(target)
+	if(!(istype(tturf, /turf/simulated)))
+		return
 	if(get_dist(src, target) <= 7)//Screen range check, so you can't get tentacle'd offscreen
 		visible_message("<span class='warning'>The [src.name] digs its tentacles under [target.name]!</span>")
 		new /obj/effect/goliath_tentacle/original(tturf)
 		ranged_cooldown = ranged_cooldown_cap
-		icon_state = icon_aggro
+		if(stat)
+			icon_state = icon_dead
+		else
+			icon_state = icon_aggro
```

Comment:
Simple animals already have checks for not openingfire while dead, this is for if they die in the middle of opening fire